src/cpp/src/continuous_batching_impl.hpp

```diff
@@ -26,18 +26,65 @@ class ContinuousBatchingPipeline::ContinuousBatchingImpl : public ContinuousBatc
 
     static const size_t AVG_CACHE_USAGE_WINDOW_SIZE_IN_STEPS = 1000;
     std::deque<float> m_previous_step_cache_usages;
+    
+    // flag to enable validation mode for sampler
+    bool m_is_validation_mode_enabled = false;
```

Comment:
we could avoid this flag validation mode is enabled via Sampler ctor as last optional argument
and `ContinuousBatchingForSpeculativeDecodingImpl ` will re-create samplers created by base class.

Reply:
we are reusing the flag to call step from base pipeline